File: 24/Ricerca.py
from Progetti import Progetti

class Ricerca(Progetti):
    def __init__(self, cod, tipo, titolo, nomeCognomeCoordinatore, organizzazione, importoTotaleInMilioniDiEuro,codiceArgomento,numeroPartner) -> None:
        super().__init__(cod, tipo, titolo, nomeCognomeCoordinatore, organizzazione, importoTotaleInMilioniDiEuro)
        self.codiceArgomento = codiceArgomento
        self.numeroPartner   = numeroPartner

    #print("tipo, titolo, codice, coordinatore, organizzazione, argomento,partner, aziende, importo totale in migliaia di euro")  
    def __str__(self) -> str:
        return self.tipo+"\t"+self.titolo+"\t"+str(self.cod)+"\t"+self.nomeCognomeCoordinatore+"\t"+self.organizzazione+"\t"+str(self.codiceArgomento)+"\t"+str(self.numeroPartner)+"\t"+"-"+"\t"+str(self.importoTotaleInMilioniDiEuro)

File: simu11/src/Ricerca.java
public class Ricerca extends Progetto {
    String codice_argomento;
    int numero_di_partner;

    public Ricerca(int codice, String tipo, String titolo_progetto, String nomeCognome_coordinatore, String orgaString,
            double importo_totale_in_milioni_di_euro, String codice_argomento, int numero_di_partner) {
        super(codice, tipo, titolo_progetto, nomeCognome_coordinatore, orgaString, importo_totale_in_milioni_di_euro);
        this.codice_argomento = codice_argomento;
        this.numero_di_partner = numero_di_partner;
    }
    @Override
    public String toString() {

        return tipo+" "+titolo_progetto+" "+codice+" "+nomeCognome_coordinatore+" "+orgaString+" "+codice_argomento+" "+numero_di_partner+" - "+(importo_totale_in_milioni_di_euro *1000);
    }

}
